src/app/components/client/homepage.tsx

```diff
@@ -38,7 +50,46 @@ const Homepage = ({ data, batch }: { data: HomepageData; batch: Array<HomepageEn
       return batch.filter((batch) => batch.dev_name === LIGHTNINGDEV);
     }
     return batch;
-  }, [mailingListSelection, batch]);
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [mailingListSelection, batch, batch.length]);
+
+  const outsideCount = count;
+  const outSideServerCount = serverCount;
+
+  const getNextBatch = async () => {
+    setloading(true);
+
+    try {
+      const countValue =
+        outsideCount === 1 && outsideCount <= serverCount[serverCount.length - 1]
+          ? count + serverCount[serverCount.length - 1]
+          : outsideCount === 1 && outSideServerCount[outSideServerCount.length - 1] >= 2
+          ? count + 1
+          : count;
+
+      if (
+        (outsideCount === 1 && outsideCount <= serverCount[serverCount.length - 1]) ||
+        (outsideCount === 1 && outSideServerCount[outSideServerCount.length - 1] >= 2)
+      ) {
+        setCount((c) => c + serverCount[serverCount.length - 1]);
+      }
+
+      const res = await next(countValue);
+      const { batch: currentBatch, count: funcCount } = res;
```

Comment:
res is a promise. Are we sure it always resolves? Or we need to add a check maybe?